File: python/algorithms/binary_search.py
def binarySearch(arr, target):
    low = 0
    high = len(arr)-1
    while(low<high):
        mid = (low+high)//2
        if(arr[mid]==target):
            return mid
        elif(arr[mid]<target):
            high=mid-1
        else:
            low =mid+1
    return -1

r = binarySearch([1,2,3,4],2)
print(r)

File: nodejs/binarySearch.js
function binarySearch(arr,target){
    let low=0;
    let high = arr.length-1;
    while(low<high){
        const mid = Math.floor((low+high)/2);
        if(arr[mid]===target) return mid;
        else if(arr[mid]<target) low=mid+1;
        else high=mid+1;
    }
    return -1;
}
let arr = [1,2,3,4]
let r = binarySearch(arr,2);
console.log(r);